public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -28,9 +28,68 @@ import {
 } from '../../../store';
 import { enrichPresetWorkflowWithUiMetadata } from './utils';
 import { getDataSourceId } from '../../../utils';
+import semver from 'semver';
+import { DataSourceAttributes } from '../../../../../../src/plugins/data_source/common/data_sources';
+import { getSavedObjectsClient } from '../../../../public/services';
 
 interface NewWorkflowProps {}
 
+export const getEffectiveVersion = async (
+  dataSourceId: string | undefined
+): Promise<string> => {
+  try {
+    // Don't make the API call if no dataSourceId
+    if (dataSourceId === undefined) {
+      console.log('cannot find data source');
+      throw new Error('Data source is required');
+    }
+
+    const dataSource = await getSavedObjectsClient().get<DataSourceAttributes>(
+      'data-source',
+      dataSourceId
+    );
+    const version = dataSource?.attributes?.dataSourceVersion || '2.17.0';
+    return version;
+  } catch (error) {
+    console.error('Error getting version:', error);
+    return '2.17.0';
+  }
+};
+
+const filterWorkflowsByVersion = async (
```

Comment:
nit: something like `filterWorkflowPresetsByVersion`/`filterPresetsByVersion`/`filterTemplatesByVersion` may be better name

Reply:
addressed on line 63